src/metabase/driver/sql_jdbc/sync/describe_database.clj

```diff
@@ -70,7 +70,9 @@
     ;; attempting to execute the SQL statement will throw an Exception if we don't have permissions; otherwise it will
     ;; truthy wheter or not it returns a ResultSet, but we can ignore that since we have enough info to proceed at
     ;; this point.
-    (.execute stmt)))
+    (doto stmt
+      (.setQueryTimeout 10)
```

Comment:
What happens if this throws a NotImplementedException?

Reply:
from @snoe :

Looking at the docs, I think they MUST implement https://docs.oracle.com/javase/8/docs/api/java/sql/Statement.html#setQueryTimeout-int- vs https://docs.oracle.com/javase/8/docs/api/java/sql/Statement.html#cancel--

we feel comfortable relying on these being implemented.